server/cli/src/generate_typescript_types.rs

```diff
@@ -0,0 +1,85 @@
+use anyhow::{anyhow, Result};
+use repository::{PluginDataFilter, PluginDataRow, StorePreferenceRow};
+use service::backend_plugin::types::{amc, graphql_query, transform_request_requisition_lines};
+use std::process::Command;
+use ts_rs::TS;
+
+#[derive(TS)]
+#[allow(unused)]
+struct Function<I: TS, O: TS> {
+    input: I,
+    output: O,
+}
+
+#[derive(TS)]
+#[allow(unused)]
+struct PluginTypes {
+    // Fields here must match PluginTypes in backend_plugin_row repository
+    average_monthly_consumption: Function<amc::Input, amc::Output>,
+    transform_request_requisition_lines: Function<
+        transform_request_requisition_lines::Input,
+        transform_request_requisition_lines::Output,
+    >,
+    graphql_query: Function<graphql_query::Input, graphql_query::Output>,
+    // Extra types to expose, not directly related to plugin interface
+    // like for input or output of global methods
+    get_store_preferences: StorePreferenceRow,
+    get_plugin_data: Function<PluginDataFilter, Vec<PluginDataRow>>,
+}
+
+pub fn generate_typescript_types() -> Result<()> {
+    println!("Running test to generate TypeScript types...");
```

Comment:
I think should be `info!` flag?

Reply:
Thanks Ferg, applied in https://github.com/msupply-foundation/open-msupply/pull/7358/commits/10905fdbb53faaec75e4b0b266b84124945df538